modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java

```diff
@@ -0,0 +1,318 @@
+
+    ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(workflowInstance,
+            Configuration.none, Configuration.one,
+            Configuration.many, Configuration.one);
+    MediaPackageElementFlavor sourceFlavor = tagsAndFlavors.getSingleSrcFlavor();
+
+    Track[] tracks = mediaPackage.getTracks(sourceFlavor);
+    if (tracks.length == 0) {
+      throw new WorkflowOperationException(
+              String.format("No tracks with source flavor '%s' found for transcription", sourceFlavor));
+    }
+
+    logger.info("Found {} track(s) with source flavor '{}'.", tracks.length, sourceFlavor);
+
+    // Get the information in which language the audio track should be
+    String languageCode = getMediaPackageLanguage(mediaPackage, workflowInstance);
+
+    // How to save the subtitle file? (as attachment, as track...)
+    AppendSubtitleAs appendSubtitleAs = howToAppendTheSubtitles(mediaPackage, workflowInstance);
+
+    for (Track track : tracks) {
+      createSubtitle(track, languageCode, mediaPackage, tagsAndFlavors, appendSubtitleAs);
+    }
+
+    logger.info("Text-to-Speech workflow operation for media package {} completed", mediaPackage);
+    return createResult(mediaPackage, WorkflowOperationResult.Action.CONTINUE);
+  }
+
+  /**
+   * Creates the subtitle file for a track and appends it to the media package.
+   *
+   * @param track The track from which the subtitles are created.
+   * @param languageCode The language of the track.
+   * @param parentMediaPackage The media package where the track is located.
+   * @param tagsAndFlavors Tags and flavors instance (to get target flavor information)
+   * @param appendSubtitleAs Tells how the subtitles file has to be appended.
+   * @throws WorkflowOperationException Get thrown if an error occurs.
+   */
+  private void createSubtitle(Track track, String languageCode, MediaPackage parentMediaPackage,
+          ConfiguredTagsAndFlavors tagsAndFlavors, AppendSubtitleAs appendSubtitleAs)
+          throws WorkflowOperationException {
+
+    // Start the transcription job, create subtitles file
+    URI trackURI = track.getURI();
+    Job job;
+    logger.info("Generating subtitle for '{}'...", trackURI);
+    try {
+      job = speechToTextService.transcribe(trackURI, languageCode);
+    } catch (SpeechToTextServiceException e) {
+      throw new WorkflowOperationException(
+              String.format("Generating subtitles for '%s' in media package '%s' failed",
+                      trackURI, parentMediaPackage), e);
+    }
+
+    if (!waitForStatus(job).isSuccess()) {
+      throw new WorkflowOperationException(
+              String.format("Speech-to-Text job for media package '%s' failed", parentMediaPackage));
+    }
+
+    // subtitles file is generated now, put it into the media package
+    try {
+      URI output = new URI(job.getPayload());
+      String id = UUID.randomUUID().toString();
+      InputStream in = workspace.read(output);
```

Comment:
Close the `InputStream`
```suggestion
      try (InputStream in = workspace.read(output)) { ...
```